### ISSUE TITLE
[FEAT] Migrate decorators to new ES Decorators

### ISSUE THREAD

**Issue by JaimeGensler** [JaimeGensler/thyseus#3]
New impl should be class auto-accessor decorators, but we could open a discussion on also supporting plain getters / setters.

This will be a breaking change that requires consumers to remove "experimentalDecorators" from their tsconfig and use the new `accessor` keyword instead of declare.

Last I looked into this, there were some issues with support particularly for prettier and some bundlers. If those aren't resolved yet, this may need to be delayed.

**Comment by 3mcd** [comment]
Link to explanation of auto-accessors, since it took me a bit to find a reference to them: https://2ality.com/2022/10/javascript-decorators.html#auto-accessors

**Comment by JaimeGensler** [comment]
Blocked by esbuild (used by Vite, our recommended bundler) not allowing `accessor` keyword.
Also on hold while serializing/deserializing objects is being investigated as an alternative to accessors.
